packages/graph-engine/src/nodes/search/findFirst.ts

```diff
@@ -0,0 +1,93 @@
+import {
+	BooleanSchema,
+	NumberSchema,
+	StringSchema
+} from '../../schemas/index.js';
+import { INodeDefinition, ToInput, ToOutput } from '../../index.js';
+import { Node } from '../../programmatic/node.js';
+import { Operator } from '../../schemas/operators.js';
+import { arrayOf } from '../../schemas/utils.js';
+
+export default class NodeDefinition extends Node {
+	static title = 'Find First Match';
+	static type = 'studio.tokens.search.findFirst';
+	static description =
+		'Finds the first array element that matches a comparison condition with a target value';
+
+	declare inputs: ToInput<{
+		array: any[];
+		target: number;
+		operator: Operator.GREATER_THAN | Operator.LESS_THAN;
+	}>;
+
+	declare outputs: ToOutput<{
+		value: any;
+		index: number;
+		found: boolean;
+	}>;
+
+	constructor(props: INodeDefinition) {
+		super(props);
+
+		this.addInput('array', {
+			type: {
+				...arrayOf(NumberSchema),
+				description: 'Array of numbers to search through'
+			}
+		});
+
+		this.addInput('target', {
+			type: {
+				...NumberSchema,
+				description: 'Target value to compare against'
+			}
+		});
+
+		this.addInput('operator', {
+			type: {
+				...StringSchema,
+				enum: [Operator.GREATER_THAN, Operator.LESS_THAN],
+				default: Operator.GREATER_THAN,
+				description: 'Comparison operator to use (greater than or less than)'
+			}
+		});
+
+		this.addOutput('value', {
+			type: {
+				...NumberSchema,
+				description: 'The first matching value found (null if no match)'
+			}
+		});
+
+		this.addOutput('index', {
+			type: {
+				...NumberSchema,
+				description: 'Index of the first matching value (-1 if no match)'
+			}
+		});
+
+		this.addOutput('found', {
+			type: {
+				...BooleanSchema,
+				description: 'Whether a matching value was found'
+			}
+		});
+	}
+
+	execute(): void {
+		const { array, target, operator } = this.getAllInputs();
+
+		const comparisonFn =
```

Comment:
do we need some kind of input validation here?  if input is anything other than greater than, it would check for less than.

Reply:
We shouldn't as it should be staticly known via the typescript declare typing and for the graph editing experience through the JSON schema